### PR TITLE
feat: add event streaming

### PR DESCRIPTION
**Pull request by thedadams**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -208,6 +256,50 @@ async function streamExecFile() {
...
+
+        stdout.on('data', data => {
+            console.log(`system: ${data}`);
+        });
+
+        stderr.on('data', data => {
+            console.log(`system: ${data}`);
+        });
+
+		events.on('data', data => {
```
Comment (cloudnautique) [inline README.md]: Formatting is a bit weird.

File: README.md [hunk 2]
```diff
@@ -168,6 +168,54 @@ async function streamExec() {
...
+
+const opts = {
+    cache: false,
+};
+
+const t = new gptscript.Tool({
+    instructions: "who was the president of the united states in 1928?"
+});
+
+async function streamExec() {
```
Comment (cloudnautique) [inline README.md]: Function should be streamExecWithEvents?

File: src/exec.js [hunk 3]
```diff
@@ -1,14 +1,16 @@
-const childProcess = require('child_process');
+const childProcess = require('child_process'),
+    net = require('net'),
```
Comment (cloudnautique) [inline src/exec.js]: Is this idiomatic?

File: README.md (hunk 1 above)
Comment (thedadams) [reply README.md]: I think I got the tabs vs spaces figured out here.

File: README.md (hunk 2 above)
Comment (thedadams) [reply README.md]: Yes, fixed.

File: src/exec.js (hunk 3 above)
Comment (thedadams) [reply src/exec.js]: Probably not. Changed it to be as in the other files.

**Review by cloudnautique — APPROVED** [review]

**Pull request merged**